utils/keycodes.js

```diff
@@ -10,3 +16,137 @@ export const DOWN = 40;
+ */
+export function keyboardShortcut( keys, _isMacOS = isMacOS ) {
+	const isMac = _isMacOS();
+
+	const alt = isMac ? '⌥' : 'Alt';
+	const meta = isMac ? '⌃' : '⊞';
+	const primary = isMac ? '⌘' : 'Ctrl';
+	const shift = isMac ? '⇧' : 'Shift';
+
+	const replacementKeyMap = {
+		[ ALT ]: alt,
+		[ PRIMARY ]: primary,
+		[ META ]: meta,
+		[ SHIFT ]: shift,
+	};
+
+	const joinCharacter = isMac ? '' : '+';
+
+	return keys
+		.replace( /\s/g, '' )
+		.split( '+' )
+		.map( ( key ) => {
+			return replacementKeyMap.hasOwnProperty( key ) ?
+				replacementKeyMap[ key ] : key;
+		} )
+		.join( joinCharacter );
+}
+
+/**
+ * Create an access key combonation shortcut based on a single character.
+ *
+ * Access key combo is: Command+Alt on MacOS; Shift+Alt everywhere else.
+ * This will output Shift+Alt+G on Windows when "G" is supplied as an argument.
+ *
+ * @param {string} character The character for the access combination.
+ * @param {Object} _isMacOS  isMacOS function by default; used for DI testing.
+ *
+ * @return {string}          The keyboard shortcut.
+ */
+export function accessShortcut( character, _isMacOS = isMacOS ) {
+	const keyCombo = _isMacOS() ? `${ META }+${ ALT }` : `${ SHIFT }+${ ALT }`;
+	return keyboardShortcut( `${ keyCombo }+${ character }`, _isMacOS );
+}
+
+/**
+ * Create a Meta shortcut based on a single character.
+ *
+ * This will output Windows+G on Windows when "G" is supplied as an argument.
+ * This will output Control+G on MacOS when "G" is supplied as an argument.
+ *
+ * @param {string} character The character for the key command.
+ * @param {Object} _isMacOS  isMacOS function by default; used for DI testing.
+ *
+ * @return {string}          The keyboard shortcut.
+ */
+export function metaShortcut( character, _isMacOS = isMacOS ) {
+	return keyboardShortcut( `${ META }+${ character }`, _isMacOS );
+}
+
+/**
+ * Create a modifier shortcut based on a single character.
+ *
+ * This will output Ctrl+G on Windows when "G" is supplied as an argument.
+ * This will output Command+G on MacOS when "G" is supplied as an argument.
+ *
+ * @param {string} character The character for the key command.
+ * @param {Object} _isMacOS  isMacOS function by default; used for DI testing.
+ *
+ * @return {string}          The keyboard shortcut.
+ */
+export function primaryShortcut( character, _isMacOS = isMacOS ) {
+	return keyboardShortcut( `${ PRIMARY }+${ character }`, _isMacOS );
+}
+
+/**
+ * Create a modifier+access shortcut based on a single character.
+ *
+ * This will output Ctrl+Shift+Alt+G on Windows when "G" is supplied as an argument.
+ * This will output Option+Shift+Command+G on MacOS when "G" is supplied as an argument.
+ *
+ * @param {string} character The character for the key command.
+ * @param {Object} _isMacOS  isMacOS function by default; used for DI testing.
+ *
+ * @return {string}          The keyboard shortcut.
+ */
+export function primaryAccessShortcut( character, _isMacOS = isMacOS ) {
```

Comment:
Could we drop this combination in favour of the standard access combination that is now called `primaryAltShortcut`?

Reply:
That'll change an existing keyboard shortcut: https://github.com/WordPress/gutenberg/blob/add/editable-shortcuts/edit-post/keyboard-shortcuts.js#L8

Which I presume has many modifiers as it isn't common. But we could call it tertiary. I am now realising I made a terrible API by surfacing its underlying implementation.

Reply:
I think we should change it be an access shortcut. It was just a random combination that didn't clash with anything.

Reply:
See https://github.com/WordPress/gutenberg/pull/3755#issuecomment-358300913 for history on that.

Reply:
I just tried Control + Alt + M on Mac and it outputs a unicode character in Gutenberg:

![2018-05-03 00 23 52](https://user-images.githubusercontent.com/90871/39554018-5ec6d5a0-4e68-11e8-9a01-adbc51876b29.gif)

But Shift + Alt works on MacOS. I guess it would be confusing to name this "access" when its rules are different… but I think just using Shift on both platforms would work. It worked in MacOS:

![2018-05-03 00 25 52](https://user-images.githubusercontent.com/90871/39554117-ea15fa6e-4e68-11e8-9677-e58c6fcffedb.gif)

Reply:
> I just tried Control + Alt + M on Mac and it outputs a unicode character in Gutenberg

Do you meant `shift+alt+m`?

That's why it is `ctrl+alt` on Mac and `shift+alt` on Windows, because there are conflicts otherwise.

Reply:
And I believe `ctrl+alt` has some conflicts on some Linux distributions.
Additionally we can't conflict with any of these: https://en.wikipedia.org/wiki/Access_key

Reply:
The docs say `access` is `ctrl+alt` which means changing the editor toggle to just `access` instead of its previous `primary + access` means it's `ctrl+alt+m` on MacOS instead of `command+ctrl+alt+m`.

`ctrl+alt+m` was outputting the Mu/µ symbol for me on MacOS. (I'm on a US keyboard, in case that matters.)

Reply:
Hm, normally character input on Mac is `LETTER`, `Shift+LETTER`, `Alt+ LETTER` and `Shift+Alt+LETTER`. I'll test a bit tomorrow then.

Reply:
Would be nice to have a combo that is the same on all OS though. :)

Reply:
Ancient history: https://core.trac.wordpress.org/ticket/29558

Reply:
Oddly enough the Mu doesn't appear *outside of Gutenberg*. If I type Ctrl+Alt+M into this text field it seems like an invisible unicode character is inserted instead.

Looks like Cmd+Alt on Mac will work, though it's also the shortcut to enable responsive design mode in Firefox so that's a bit annoying. 😆

Maybe this needs to stay a three-character combo, just to be safe?

Reply:
If there's one clash in that namespace, there's likely to be more. Would be nice to find a maximum two-modifier namespace that can be used in WordPress.

I wonder if we can adopt one of https://en.wikipedia.org/wiki/Access_key...

Curiously enough it list `Ctrl+Alt` as reserved on Firefox on Mac for `accesskey`, so I wonder why it decided to use that.

I'm fine with just picking _one_ non primary combination in this PR and revise it later.